File: level3/네트워크.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(int n, vector<vector<int>> computers) {
    vector<bool> visit(n,false);
    
    int answer = 0;
    for(int i = 0; i < n; i++){
        if(visit[i]){
            continue;
        }
        answer++;
        
        queue<int> q;
        q.push(i);
        visit[i] = true;
        while(!q.empty()){
            int com = q.front();
            q.pop();
            for(int j = 0; j < n; j++){
                if(computers[com][j] == true && !visit[j]){
                    q.push(j);
                    visit[j] = true;
                }
            }
        }
    }
    
    return answer;
}

File: level3/네트워크.py
def solution(n, computers):
    answer = 0

    visit = [0]*n
    stack = []
    for i in range(n):
        if visit[i] == 0:
            visit[i] = 1
            for j in range(n):
                if computers[i][j] == 1:
                    stack.append(j)
                    computers[i][j] = 0
                    computers[j][i] = 0    
                    
            while stack:
                num = stack.pop()
                if visit[num] == 0:
                    visit[num] = 1
                    for j in range(n):
                        if computers[num][j] == 1:
                            stack.append(j)
                            computers[num][j] = 0
                            computers[j][num] = 0                            
            answer += 1
                                
    return answer
